Python/bytecodes.c

```diff
@@ -2208,32 +2213,58 @@ dummy_func(
             _LOAD_ATTR_MODULE_FROM_KEYS +
             unused/5;
 
-        op(_CHECK_ATTR_WITH_HINT, (owner -- owner)) {
+        op(_CHECK_ATTR_WITH_HINT, (owner -- owner, dict: PyDictObject *)) {
             PyObject *owner_o = PyStackRef_AsPyObjectBorrow(owner);
 
             assert(Py_TYPE(owner_o)->tp_flags & Py_TPFLAGS_MANAGED_DICT);
-            PyDictObject *dict = _PyObject_GetManagedDict(owner_o);
-            EXIT_IF(dict == NULL);
-            assert(PyDict_CheckExact((PyObject *)dict));
+            PyDictObject *dict_o = _PyObject_GetManagedDict(owner_o);
+            EXIT_IF(dict_o == NULL);
+            assert(PyDict_CheckExact((PyObject *)dict_o));
+            dict = dict_o;
         }
 
-        op(_LOAD_ATTR_WITH_HINT, (hint/1, owner -- attr, null if (oparg & 1))) {
-            PyObject *owner_o = PyStackRef_AsPyObjectBorrow(owner);
+        op(_LOAD_ATTR_WITH_HINT, (hint/1, owner, dict: PyDictObject * -- attr, null if (oparg & 1))) {
             PyObject *attr_o;
+            if (!LOCK_OBJECT(dict)) {
+                DEAD(dict);
+                POP_DEAD_INPUTS();
```

Comment:
This seems rather obscure. I see why you need to pop `dict`, but `DEAD(dict); POP_DEAD_INPUTS()` seems a rather convoluted way to do this.

Since you adding a new code generator macro, why not `POP(dict)`?
I think that most people would expect that an explicitly popped value is no longer live, so there should be no need for a kill and a pop.

Reply:
Sure, I can look at that. `POP` is already taken by a macro in `ceval_macros.h`, but we could use `POP_INPUT`.

Reply:
Aside:
We shouldn't be using `POP`, as the code generator should handle pops. We can remove `POP` in another PR.